libraries/botframework-connector/src/auth/certificateServiceClientCredentialsFactory.ts

```diff
@@ -37,23 +40,55 @@ export class CertificateServiceClientCredentialsFactory extends ServiceClientCre
         certificatePrivateKey: string,
         tenantId?: string,
         x5c?: string
+    );
+
+    /**
+     * Initializes a new instance of the CertificateServiceClientCredentialsFactory class.
+     *
+     * @param appId Microsoft application Id related to the certificate.
+     * @param x5c Value that enables application developers to achieve easy certificates roll-over in Azure AD
+     * set this parameter to send the public certificate (BEGIN CERTIFICATE) to Azure AD, so that Azure AD can use it to validate the subject name based on a trusted issuer policy.
+     * @param certificatePrivateKey A PEM encoded certificate private key.
+     * @param tenantId Optional. The oauth token tenant.
+     */
+    constructor(appId: string, x5c: string, certificatePrivateKey: string, tenantId?: string);
+
+    /**
+     * @internal
+     */
+    constructor(
+        appId: string,
+        certificateThumbprintOrx5c: string,
+        certificatePrivateKey: string,
+        tenantId?: string,
+        x5c?: string
     ) {
         super();
+
         ok(appId?.trim(), 'CertificateServiceClientCredentialsFactory.constructor(): missing appId.');
-        ok(
-            certificateThumbprint?.trim(),
-            'CertificateServiceClientCredentialsFactory.constructor(): missing certificateThumbprint.'
-        );
         ok(
             certificatePrivateKey?.trim(),
             'CertificateServiceClientCredentialsFactory.constructor(): missing certificatePrivateKey.'
         );
 
+        if (certificateThumbprintOrx5c.includes('-----BEGIN CERTIFICATE-----')) {
+            this.x5c = certificateThumbprintOrx5c;
+            ok(
+                certificateThumbprintOrx5c.trim(),
+                'CertificateServiceClientCredentialsFactory.constructor(): missing x5c.'
+            );
```

Comment:
now looking at this, this validation will not be triggered anymore, because will fail if the certificateThumbprintOrx5c is empty, which will not be the case.

And add in the else in the validation the `or x5c` like you had before because we dont know if it is the thumbprint or the x5c